### PR TITLE
Convert jumpsuits, jumpskirts and sneakers to GAGS

### PR DESCRIPTION
**Pull request by celotajstg**
## About The Pull Request

By @Qustinnus in #58112:
image

I got you.

So this PR converts jumpsuits, jumpskirts and sneakers to be generated using GAGS.
image image

Object steps are shown below (very similar steps are for worn and inhand states):
image 
image

One thing I have to note however is that old inhands are not removed due to being used by other clothing items, and that this PR does not affect rainbow and ancient jumpsuits.

## Why It's Good For The Game

More consistent jumpsuits, jumpskirts and sneakers, as well as less space used.

## Changelog
:cl: Celotajs
refactor: Jumpsuits, jumpskirts and sneakers are now generated using GAGS
/:cl:

### CODE REVIEW

**Comment by celotajstg** [comment]
Just in case - the integration test failure is to be expected. I'm waiting for #58487 to get merged, and then I'll do a rebase which will fix this.

**Comment by Fikou** [comment]
why are you removing the ancient jumpsuit icon? its supposed to be 100% identical to the r4407 sprites

**Comment by celotajstg** [comment]
Its icon was pretty much identical to the rest, but I'll add an exception for it, will just make it use the legacy grey inhand too then.

**Comment by ninjanomnom** [comment]
suit

I'm pretty sure you didn't intend for this change

**Comment by celotajstg** [comment]
Okay, all oversights and issues should now be fixed.

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/modules/clothing/under/color.dm [hunk 1]
```diff
@@ -56,163 +67,133 @@
...
-	icon_state = "grey_skirt"
-	inhand_icon_state = "gy_suit"
+	greyscale_colors = "#b3b3b3"
 
 /obj/item/clothing/under/color/grey/ancient
 	name = "ancient jumpsuit"
 	desc = "A terribly ragged and frayed grey jumpsuit. It looks like it hasn't been washed in over a decade."
 	icon_state = "grey_ancient"
+	inhand_icon_state = "gy_suit"
 	can_adjust = FALSE
```
Comment (Fikou) [inline code/modules/clothing/under/color.dm:82]: this inherits greyscale_colors and config doesnt it? rainbow jumpsuit inherits config too

Comment (celotajstg) [reply code/modules/clothing/under/color.dm:82]: Yep, it does, but due to no colors it does nothing at all

**Comment by celotajstg** [comment]
Will fix the error in a few hours

**Comment by celotajstg** [comment]
So it turned out GAGS icons have a default icon size of 0x0. Declaring the icon size during vendor sprite sheet generation as 32x32 resolved the issue.

**Comment by celotajstg** [comment]
Updated the description.
The changes are ready.

File: code/modules/clothing/under/color.dm (hunk 1 above)
Comment (celotajstg) [reply code/modules/clothing/under/color.dm]: Decided to make a more clear declaration that prevents config inheritance

**Comment by Dorsisdwarf** [comment]
This shouldnt have any issues with that memory problem the servers were having with grey scaled items last week should it?

**Comment by celotajstg** [comment]
@Dorsisdwarf I believe there will be no issues with this PR.

**Review by Twaticus — APPROVED** [review]
radical

**Comment by ninjanomnom** [comment]
> @Dorsisdwarf  
> 
> This shouldnt have any issues with that memory problem the servers were having with grey scaled items last week should it?

This was fixed in #58378

File: code/modules/asset_cache/asset_list_items.dm [hunk 2]
```diff
@@ -445,6 +451,10 @@
 			stack_trace("[item] does not have a valid icon state, icon=[icon_file], icon_state=[json_encode(icon_state)](\ref[icon_state]), icon_states=[icon_states_string]")
 			I = icon('icons/turf/floors.dmi', "", SOUTH)
 
+		if(is_gags_icon)
+			// Scales the icon to 32x32, as it is 0x0 by default. Type checks can be added here if other sizes are necessary.
+			I.Scale(32, 32)
```
Comment (ninjanomnom) [inline code/modules/asset_cache/asset_list_items.dm:456]: GAGS supports sizes other than 32x32, was converting the icon to an image not a good way of getting width and height?

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/modules/asset_cache/asset_list_items.dm (hunk 2 above)
Comment (celotajstg) [reply code/modules/asset_cache/asset_list_items.dm:456]: Haven't really got that approach to work particularly well to be honest. I think considering vendor icons are typically 32x32 this is a sensible solution, as from what I know none of the vendors use a different size.

Comment (celotajstg) [reply code/modules/asset_cache/asset_list_items.dm:456]: This approach also allows avoiding a potentially more expensive operation of converting the icon to an image that ends up not getting used later on, as the sprite sheet function accepts only icons, not images

**Comment by celotajstg** [comment]
todo: add a better fix for vendors (coordinated with #58567)

**Comment by celotajstg** [comment]
Rebased to resolve a merge conflict

**Review by ninjanomnom — CHANGES_REQUESTED** [review]
Update to the latest master instead of duplicating the sprite size fixes in here and you should be good to go

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**